src/tyr/actor.cc

```diff
@@ -51,69 +51,103 @@ void actor_t::cleanup() {
   pimpl->cleanup();
 }
 
+std::string actor_t::act(Api& api, const std::function<void()>* interrupt) {
+  if (!api.options().has_action_case())
+    throw valhalla_exception_t{106};
+
+  switch (api.options().action()) {
+    case Options::route:
+      return route("", interrupt, &api);
+    case Options::locate:
+      return locate("", interrupt, &api);
+    case Options::sources_to_targets:
+      return matrix("", interrupt, &api);
+    case Options::optimized_route:
+      return optimized_route("", interrupt, &api);
+    case Options::isochrone:
+      return isochrone("", interrupt, &api);
+    case Options::trace_route:
+      return trace_route("", interrupt, &api);
+    case Options::trace_attributes:
+      return trace_attributes("", interrupt, &api);
+    case Options::height:
+      return height("", interrupt, &api);
+    case Options::transit_available:
+      return transit_available("", interrupt, &api);
+    case Options::expansion:
+      return expansion("", interrupt, &api);
+    case Options::centroid:
+      return centroid("", interrupt, &api);
+    case Options::status:
+      return status("", interrupt, &api);
+    default:
+      throw valhalla_exception_t{106};
+  }
+}
+
 std::string
 actor_t::route(const std::string& request_str, const std::function<void()>* interrupt, Api* api) {
   // set the interrupts
   pimpl->set_interrupts(interrupt);
+  // if the caller doesn't want a copy we'll use this dummy
```

Comment:
essentially there are two inputs to this function now, either the request string or the api object. previously the api object was just something that you could optionally get out but now it can be the primary input to this and the other functions like this. in that case we need to not overwrite (via swap) but rather mutate it accordinly instead. thats why when its not passed in by the user but the json request string is used we use the dummy instead.